generators/pkg/generator/generate.go

```diff
@@ -194,6 +209,148 @@ func GeneratePipeline(github *GitHub) (*v1beta1.Pipeline, error) {
 	return pipeline, nil
 }
 
+// Generate the pipelinerun with the given generated pipeline
+func GeneratePipelineRun(p *v1beta1.Pipeline, g *GitHub) *v1beta1.PipelineRun {
+	if p.Namespace == "" {
+		p.Namespace = "default"
+	}
+
+	var value int64
+	var format resource.Format
+	if g.Spec.Storage != "" {
+		diskSize := resource.MustParse(g.Spec.Storage)
+		value = diskSize.Value()
+		format = diskSize.Format
+	} else {
+		value = 1024 * 1024 * 1024
+		format = resource.BinarySI
+	}
+
+	if g.Spec.Revision == "" {
+		g.Spec.Revision = "master"
+	}
+
+	return &v1beta1.PipelineRun{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: v1beta1.SchemeGroupVersion.String(),
+			Kind:       "PipelineRun",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: p.Namespace,
+			Name:      p.Name + "-run",
+			Labels:    p.Labels,
+		},
+		Spec: v1beta1.PipelineRunSpec{
+			PipelineRef: &v1beta1.PipelineRef{
+				Name: p.Name,
+			},
+			Params: []v1beta1.Param{
+				{
+					Name: "gitrepositoryurl",
+					Value: v1beta1.ArrayOrString{
+						Type:      v1beta1.ParamTypeString,
+						StringVal: g.Spec.URL,
+					},
+				},
+				{
+					Name: "gitrevision",
+					Value: v1beta1.ArrayOrString{
+						Type:      v1beta1.ParamTypeString,
+						StringVal: g.Spec.Revision,
+					},
+				},
+			},
+			Workspaces: []v1beta1.WorkspaceBinding{
+				{
+					Name: p.Spec.Workspaces[0].Name,
+					VolumeClaimTemplate: &corev1.PersistentVolumeClaim{
+						Spec: corev1.PersistentVolumeClaimSpec{
+							AccessModes: []corev1.PersistentVolumeAccessMode{
+								corev1.ReadWriteOnce,
+							},
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									"storage": *resource.NewQuantity(value, format),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// GenerateTaskRun generates Tekton TaskRun
+// from simplified Github configs.
+func GenerateTaskRun(g *GitHub) *v1beta1.TaskRun {
+	if g.Namespace == "" {
+		g.Namespace = "default"
+	}
+
+	labels := g.Labels
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+	labels["generator.tekton.dev"] = g.Name
+
+	var value int64
+	var format resource.Format
+	if g.Spec.Storage != "" {
+		diskSize := resource.MustParse(g.Spec.Storage)
+		value = diskSize.Value()
+		format = diskSize.Format
+	} else {
+		value = 1024 * 1024 * 1024
+		format = resource.BinarySI
+	}
+
+	if g.Spec.ServiceAccountName == "" {
+		g.Spec.ServiceAccountName = "default"
+	}
+
```

Comment:
This seems duplicative with PipelineRun. Can we reuse code / pull this into a func?